services/director-v2/src/simcore_service_director_v2/api/routes/computations.py

```diff
@@ -164,10 +165,14 @@ async def _get_project_metadata(
 
     parent_node_id = NodeID(current_project_metadata["node_id"])
     parent_node_idstr = NodeIDStr(f"{parent_node_id}")
-    parent_project_id = await project_repo.get_project_id_from_node(parent_node_id)
-    parent_project = await project_repo.get_project(parent_project_id)
-    assert parent_node_idstr in parent_project.workbench
+    try:
+        parent_project_id = await project_repo.get_project_id_from_node(parent_node_id)
+        parent_project = await project_repo.get_project(parent_project_id)
+    except (ProjectNotFoundError, ProjectNodesNodeNotFound) as exc:
+        _logger.exception("Could not find project/node: %s", f"{exc}")
+        return {}
```

Comment:
Careful, just for consistency, make sure that `ProjectMetadataDict` has `total=False` i.e. all keys are optional. Otherwise some of the logic aftewards might be buggy (SEE https://peps.python.org/pep-0589/#totality)

Reply:
Good point, thanks. Indeed `ProjectMetadataDict` has `total=False`